contracts/VaultV3.vy

```diff
@@ -555,63 +571,27 @@ def _update_debt(strategy: address) -> uint256:
-          h1 = t0, h2 = t10 and h3 = t13 =>
-              current_time_delta = 3, (new)previous_harvest_time_delta = 7 (10-3), locked until t20
-          h1 = t0, h2 = t10 and h3 = t14 =>
-              current_time_delta = 4, (new)previous_harvest_time_delta = 6 (10-4), locked until t20
-          on 2nd example: h2 is getting carried into h3 (minus time delta 4) since it was previously trying to reach t20.
-          so it continues to spread the lock up to that point, and thus avoids reducing the previous distribution time.
-
-          if locked-until is unchanged, to avoid extra storage read and subtraction cost [behaves as examples below]
-          h1 = t0, h2 = t10 and h3 = t15 =>
-              current_time_delta = 5, (new)previous_harvest_time_delta = 5 locked until t20
-
-          if next total time delta is higher than previous period remaining, locked-until will increase
-          h1 = t0, h2 = t10 and h3 = t16 =>
-              current_time_delta = 6, (new)previous_harvest_time_delta = 6 locked until t22
-          h1 = t0, h2 = t10 and h3 = t17 =>
-              current_time_delta = 7, (new)previous_harvest_time_delta = 7 locked until t24
-
-          current_time_delta is the time delta between now and last_report.
-          previous_harvest_time_delta is the time delta between last_report and the previous last_report
-          previous_harvest_time_delta is assigned the higher value between current_time_delta and (previous_harvest_time_delta - current_time_delta)
-    """
-
-    # TODO: check how to solve deposit sniping for very profitable and infrequent strategy reports
-    # when there are also other more frequent strategies reducing time delta.
-    # (need to add time delta per strategy + accumulator)
-    current_time_delta: uint256 = block.timestamp - self.last_report
-    if self.previous_harvest_time_delta > current_time_delta * 2:
-      self.previous_harvest_time_delta = self.previous_harvest_time_delta - current_time_delta
-    else:
-      self.previous_harvest_time_delta = current_time_delta
-    self.last_report = block.timestamp
+def _update_profit_distribution():
+    if self.profit_distribution_rate != 0:
+        unlocked_profit: uint256 = 0 
+        if block.timestamp > self.profit_end_date:
+            unlocked_profit = (self.profit_end_date - self.profit_last_update) * self.profit_distribution_rate / MAX_BPS
+            self.profit_distribution_rate = 0
+        else:
+            unlocked_profit = (block.timestamp - self.profit_last_update) * self.profit_distribution_rate / MAX_BPS
+        self.total_debt += unlocked_profit
+        self.profit_last_update = block.timestamp
 
 
 @internal
 def _process_report(strategy: address) -> (uint256, uint256):
     assert self.strategies[strategy].activation != 0, "inactive strategy"
+
+    # Let´s update locked profit values to work with most up to date data
+    self._update_profit_distribution()
```

Comment:
I think we need to update the current state before updating the P&L. 

So before updating the `profit_distribution_rate` with a new value (due to a new profit), we need to compute the unlocked profits due to the `profit_distribution_rate` till this very moment, otherwise we would lose track of those unlocked profits. Does that make sense?

Reply:
it does make sense but you don't need to save it -- and spend gas -- as it will be overwritten in case of gains / losses. do you know what I mean?  

so I vote you move it down in the function and calculate to use it in the profit / loss recording